test/re/frida/MethodTest.java

```diff
@@ -55,14 +59,183 @@ public void replacementPropagatesExceptions() {
+    @Test
+    public void nativeReturnGenericBadgerWrapperAroundJavaLangClass() { 
+        loadScript(
+                "try {" +
+                "  var c = Java.use('re.frida.Badger');" +
+                "  var orig = c.forNameYo;" +
+                "  c.forNameYo.implementation = function () {" +
+                "    orig.call(this);" +
+                "  };" +
+                "  var test = c.forNameYo('re.frida.Badger', false, null);" +
+                "  send('ok');" +
+                "} catch(e) {" + 
+                "  send('nativeReturnGeneric: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    // this one was just hanging indefinitely during the test, but in an actual app, it was crashing
+    //! either one of those is bad.
+    /*@Test
+    public void methodInvoke() {
+        loadScript("var c = Java.use('java.lang.reflect.Method');" +
+                "var c2 = Java.use('java.lang.Class');" +
+                "try {" +
+                
+                // hook the original
+                "  var orig = c.invoke;" +
+                "  c.invoke.implementation = function (obj, ...args) { " +
+                "    orig.call(this, obj, args);" +
+                "  };" +
+                
+                // now call it and see what happens
+                "  var cl = c2.forName('re.frida.Badger');" +
+                "  var method = cl.getMethod('returnZero', 'int');" +
+                "  var ret = method.invoke();" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('Method.invoke: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }*/
+    
+    @Test
+    public void loadWorks() {
+        loadScript("var c = Java.use('java.lang.System');" +
+                "try {" +
+                "  var orig = c.load;" +
+                "  c.load.implementation = function (s) { " +
+                "    orig.call(this,s);" +
+                "  };" +
+                "  c.load('/system/lib/libc.so')" +
+                "  send('ok');" +
+                "} catch (e) {" + 
+                "  send('System.load: ' + e);" + 
+                "}"
+                );
+        assertEquals("ok", script.getNextMessage());
+    }
+    
+    @Test
+    public void runtimeLoadLibrary() {        
+        loadScript("var c = Java.use('java.lang.Runtime');" +
+                "try {" +
+                "  var orig = c.loadLibrary.overload('java.lang.String');" +
+                "  c.loadLibrary.overload('java.lang.String').implementation = function (s) {" +
+                "    orig.call(this,s);" +
+                "  };" +
+                
+                // now look up the function again and call it
+                "  var now = c.loadLibrary.overload('java.lang.String');" +
+                "  now.call(this, '/system/lib/libc.so')" +
```

Comment:
`this` is bound to the wrong variable here – it should be bound to `c` instead of `this`.